### ISSUE TITLE
when portal fails to ask CH whether an account exists, portal assumes the account doesn't exist

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-portal#882]
To reproduce on a CHAPI-style portal/CH:
 * Login to the portal web UI using an existing account, and verify that you see the map splash screen
 * On the portal/CH host, kill the `/usr/sbin/fcgi-pm -k start` process
 * Reload the portal web UI page.
What i see is that i'm redirected to `/secure/kmactivate.php`, and if i then click "Activate", i get taken to `/secure/do-register.php` and see
```
error on identity insert: MDB2 Error: constraint violation
```

I don't know whether or not this problem exists in the portal when it talks to the proto-ch-style CH, or only when it talks to the CHAPI-style CH.

 _Imported from trac ticket [#882](http://trac.gpolab.bbn.com/proto-ch/ticket/882),  created by chaos on 11-11-2013 at 16:50, last modified: 12-05-2013 at 13:11_

**Comment by ahelsing** [comment]
Incidentally, once you've tested in this fashion, restart apache to set things right again with your CH.

_Trac comment by chaos on 11-11-2013 at 16:50_

**Comment by ahelsing** [comment]
portal/www/porta/user.php tries to load the user by EPPN. In particular:
```
  $member = ma_lookup_member_by_eppn($ma_url, Portal::getInstance(), $eppn);
  if (is_null($member)) {
    // New identity, go to activation page
    relative_redirect("kmactivate.php");

```

As you can see, if the library function returns null, then we assume this is a new user. There is no affordance for getting an error message back.

In the CHAPI `ma_client.php` there is no error checking of the return from the XMLRPC calls, specifically in `ma_lookup_members_by_identifying` (the function that does the real work for this case).


_Trac comment by ahelsing on 11-12-2013 at 10:48_

**Comment by MarshallBrinn** [comment]
Here's the plan (discussed with Aaron and Marshall 11-14-2013):

1. We add an additional field to error-text.php to differentiate normal failures from system failures (say, ?system_error=1

2. We add in code to chapi.php to check if the $result is a valid array (either has faultString or code/value/output).
    If not, we redirect to error-text with ?system_error = 1

3. In error-text.php, we dcall show_header with load_user = 0 if system_error. Also, we take the error text and htmlencode it.

This will cause us to trap normal errors in the normal way, but to trap system errors (problems with CHAP) and send an error page that doesn't have infinite regress or hanging.

I'll try to write/test this later today. 



_Trac comment by mbrinn (github user: MarshallBrinn) on 11-14-2013 at 11:32_

**Comment by MarshallBrinn** [comment]
Fixed tested and checked in git tag [a84476a06225425c593ec1640bc4e3a5f917b5aa](https://github.com/GENI-NSF/geni-portal/commit/a84476a06225425c593ec1640bc4e3a5f917b5aa). 

_Trac comment by mbrinn (github user: MarshallBrinn) on 11-21-2013 at 12:37_
